onnxruntime/test/util/default_providers.cc

```diff
@@ -224,6 +224,15 @@ std::unique_ptr<IExecutionProvider> DefaultQnnExecutionProvider() {
 #endif
 }
 
+std::unique_ptr<IExecutionProvider> DefaultQnnExecutionProviderWithOptions(const ProviderOptions& options) {
+#ifdef USE_QNN
+  return QNNProviderFactoryCreator::Create(options)->CreateProvider();
+#else
+  ORT_UNUSED_PARAMETER(options);
+  return nullptr;
+#endif
+}
```

Comment:
Should we keep this one and remove line 212-225? And update the code below to use your new method.
https://github.com/microsoft/onnxruntime/blob/6c8538f0862f213ba8df3ffde67e81024a7a810f/onnxruntime/test/providers/provider_test_utils.cc#L1263

Reply:
We would have to replicate the logic for setting the correct backend path depending on the OS (.dll vs .so). Other EPs have separate functions for creating a default with options. I'm leaning towards leaving it as is.